font/src/ft/mod.rs

```diff
@@ -529,6 +581,84 @@ impl FreeTypeRasterizer {
     }
 }
 
+fn downsample_bitmap(mut bitmap_glyph: RasterizedGlyph, fixup_factor: f64) -> RasterizedGlyph {
+    // Don't try to upscale
+    if fixup_factor >= 1.0 {
+        return bitmap_glyph;
+    }
+
+    let bitmap_buffer = if let BitmapBuffer::RGBA(bitmap_buffer) = bitmap_glyph.buf {
+        bitmap_buffer
+    } else {
+        return bitmap_glyph;
+    };
+
+    let bitmap_width = bitmap_glyph.width as usize;
+    let bitmap_height = bitmap_glyph.height as usize;
+
+    let width = (bitmap_width as f64 * fixup_factor) as usize;
+    let height = (bitmap_height as f64 * fixup_factor) as usize;
+
+    let advance_step = (1.0 / fixup_factor).ceil() as usize;
+    let mut scaled_buffer = Vec::<u8>::with_capacity(width * height * 4);
+
+    let mut source_line_index = 0;
+
+    for _ in 0..height {
+        let mut source_column_index = 0;
```

Comment:
I think I've mentioned this in the previous review, but `source_start_line` and `source_start_column` would make a lot more sense here in combination with the `source_end_line` and `source_end_column` variables.

Reply:
It's not a start it's an index. Since you can't change start, but you can change index. So, I'm not sure about it.

Reply:
Of course it's a start. It's the start of the source line range.